content/en/docs/usage/certificate.md

```diff
@@ -167,10 +167,208 @@ If you would prefer the `Secret` to be deleted automatically when the `Certifica
\ No newline at end of file
+|     Field     |      Default      |     Minimum      |        Requirement         |
+|---------------|-------------------|------------------|----------------------------|
+| `duration`    | 90 days (`2160h`) | 1 hour (`1h`)    |                            |
+| `renewBefore` | 30 days (`720h`)  | 5 minutes (`5m`) | `duration` > `renewBefore` |
+
+The remaining of this section details how cert-manager uses the `renewBefore`
+hint to decide when to attempt renewing the Certificate.
+
+[certspec]: /docs/reference/api-docs/#cert-manager.io/v1.CertificateSpec
+[notafter]: https://www.rfc-editor.org/rfc/rfc5280.html#section-4.1.2.5
+
+Let's imagine that we create a Certificate on the 1st of January. We want the
+Certificate to expire after three months, and we want cert-manager to renew it
+one month before expiry:
+
+```yaml
+apiVersion: cert-manager.io/v1
+kind: Certificate
+spec:
+  duration: 2160h          # 90 days
+  renewBefore: 360h        # 15 days
+```
+
+With `renewBefore` set to `720h`, we are expecting the renewal to happen on the
+15th of March:
+
+```diagram
+                                    duration (90 days)
+     <------------------------------------------------------------------------------>
+
+                                                                                expected
+                                                              expected         certificate
+ certificate                                                 reissuance          expiry
+   issued                                                       time              time
+     +-----------------------------------------------------------+------------------+
+   1 Jan                                                       15 Mar            31 Mar
+
+                                                                  <-----------------+
+                                                                 renewBefore (15 days)
+```
+
+After issuing the Certificate, cert-manager looks at the
+[`notAfter`](https://www.rfc-editor.org/rfc/rfc5280.html#section-4.1.2.5) field
+(inside the X.509 certificate) in order to calculate what we call the
+`actualDuration`.
+
+{{% alert title="duration may be ignored by the issuer" color="warning" %}}
+Some issuers might be configured to only issue certificates with a set expiry
+duration, so the actual duration (calculated using the X.509 certificate's
+`notAfter` field) may be different. The `duration` field is a "hint" given
+to the issuer.
+
+We call `actualDuration` the duration before expiry that was set be the issuer:
+
+```
+actualDuration = notAfter - issuanceTime
+```
+
+{{% /alert %}}
+
+cert-manager computes a "safe" maximum duration that we call `maxRenewBefore`,
+set to a third of the actual duration:
```

Comment:
At this point I get really confused. What's `maxRenewBefore`? That's not a field or a constant in the cert-manager API so I think it's going to confuse others too.

But I think I see what you're trying to highlight.

The original document says:

> Once a certificate has been issued, `cert-manager` will calculate _how long_ before expiry a cert should be renewed using the formula `min(spec.renewBefore, actual_duration / 3)` 

So imagine that I want to configure a Certificate to be renewed daily, but the CA always issues 30d certificates:
```
NotBefore: 1 Jan 00:00:00
NotAfter: 31 Jan 00:00:00
actualDuration = 30d

spec.renewBefore = 29d
actualRenewalBefore = min(spec.renewBefore, actual_duration / 3) = 10d

expectedRenewalTime = 2 Jan 00:00:00
actualRenewalTime = 21 Jan 00:00:00
```

So cert-manager is going to override my quite reasonable configuration.

Reply:
> At this point I get really confused. What's maxRenewBefore? That's not a field or a constant in the cert-manager API so I think it's going to confuse others too.

Yeah, I shouldn't have defined yet another "name", people are going to look for this in the Certificate API 😞